File: app-android/app/src/main/java/de/tutao/tutanota/offline/Entity.java
package de.tutao.tutanota.offline;

import androidx.annotation.NonNull;

@androidx.room.Entity(
		primaryKeys = {
				"listId", "elementId", "typeRef"
		}
)
public class Entity {

	/**
	 * List id of the entity. An empty string means it's not a list element
	 */
	@NonNull
	public String listId = "";

	@NonNull
	public String elementId = "";

	@NonNull
	public String typeRef = "";


	public String entityJson;
}

File: app-android/app/src/main/java/de/tutao/tutanota/offline/Entity.kt
package de.tutao.tutanota.offline

@androidx.room.Entity(primaryKeys = ["listId", "elementId", "typeRef"])
data class Entity(
	val typeRef: String,
	val listId: String,
	val elementId: String,
	val data: String
) {
	companion object {
		const val NO_LIST_ID = ""
	}
}